pyccel/ast/cudaext.py

```diff
@@ -174,6 +174,62 @@ def __init__(self, dim=None):
     def dim(self):
         return self._dim
 
+
+class CudaCopy(CudaNewArray):
```

Comment:
Any reason to have inherit from `CudaNewArray` ? shouldn't this be a `PyccelInternalFunction` instead ?

Reply:
Due to the creation of a new array. And As stated in the docstring, cudaNewArray is the base class for all the cuda functions that imply a call to allocate.

Reply:
But unless i missed something your `CudaCopy` doesn't allocate anything by itself, just copies the data from one array to another,  if your Class Inherited from `PyccelInternalFunction` for example would it change anything ?

Reply:
Yes, `CudaCopy` only copies the data from one array to a new cuda array that will be created. 
In case of `PyccelInternalFunction` as base class, I should add the instance of `CudaCopy` in the type inference which is the same as `CudaNewArray`.
I think that `CudaNewArray `as base class is wokring just fine.

Reply:
Sure that works too.